packages/core/src/state-management/anchor-resuming-service.ts

```diff
@@ -33,24 +37,37 @@ export class AnchorResumingService {
     })
   }
 
+  delay(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms))
+  }
+
   async resumeRunningStatesFromAnchorRequestStore(repository: Repository): Promise<void> {
     if (this.#shouldBeClosed) {
       throw Error('This AnchorResumingService is closed, create a new instance to resume')
     }
 
+    this.logger.imp(`Resuming polling for streams with pending anchors`)
+
+    let numRestoredStreams = 0
     let gt: StreamID | undefined = undefined
     let batch = new Array<AnchorRequestStoreListResult>()
     do {
       batch = await repository.anchorRequestStore.list(RESUME_BATCH_SIZE, gt)
-      batch.forEach((listResult) => {
+      for (const listResult of batch) {
         this.resumeQ.add(async () => {
```

Comment:
We should jitter before adding to prevent thundering herd as well.

Reply:
we sleep after adding each entry to the queue so no need to sleep before, that would just be redundant.

Reply:
Ah, then we want the jitter at https://github.com/ceramicnetwork/js-ceramic/blob/c6906048236e9b101149bc189a88ecbe27fb9234/packages/core/src/state-management/anchor-resuming-service.ts#L60 so we interrupt before the next one kicks off.

Reply:
sorry I still don't understand how that isn't redundant.  We already sleep between each stream

Reply:
we add all the requests for getting the streams to the queue, sleeping between that. We don't necessarily want to immediately readd to the queue when a request finishes, but rather give the event loop time to do other work.